sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -1666,9 +1677,14 @@ case class WidthBucket(
   }
 
   override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
-    defineCodeGen(ctx, ev, (input, min, max, numBucket) =>
-      "org.apache.spark.sql.catalyst.expressions.WidthBucket" +
-        s".computeBucketNumber($input, $min, $max, $numBucket)")
+    nullSafeCodeGen(ctx, ev, (input, min, max, numBucket) => {
```

Comment:
Sorry if I am missing anything, but with the change to `nullSafeCodeGen` here, we should be able to handle null inputs right? Why we need to change implementation of `computeBucketNumber()` as well?

Reply:
Yes, with `nullSafeCodeGen` we will be able to handle null inputs. In fact, `defineCodeGen` already used `nullSafeCodeGen`, so we were already able to handle null inputs before. The issue is that `computeBucketNumber()` can return null when none of the inputs are null. Before this change it would throw an NPE in that case because it would try to assign null to a `long`.

Reply:
Oh I see, got it. @tomvanbussel thanks for explanation.